cli/module_build.go

```diff
@@ -387,3 +392,113 @@ func jobStatusFromProto(s buildpb.JobStatus) jobStatus {
 		return jobStatusUnspecified
 	}
 }
+
+// ReloadModuleAction builds a module, configures it on a robot, and starts or restarts it.
+func ReloadModuleAction(c *cli.Context) error {
+	if !c.Bool(moduleBuildRestartOnly) {
+		return fmt.Errorf("only %s mode is supported for now", moduleBuildRestartOnly)
+	}
+	partID, err := resolvePartID(c, "/etc/viam.json")
+	if err != nil {
+		return err
+	}
+	manifest, err := loadManifestOrNil(c.String(moduleFlagPath))
+	if err != nil {
+		return err
+	}
+	vc, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	return restartModule(c, vc, partID, manifest)
+}
+
+// loadManifestOrNil doesn't throw error on missing.
+func loadManifestOrNil(path string) (*moduleManifest, error) {
+	manifest, err := loadManifest(path)
+	if err == nil {
+		return &manifest, nil
+	}
+	if os.IsNotExist(err) {
+		//nolint:nilnil
+		return nil, nil
+	}
+	return nil, err
+}
+
+// resolvePartID takes an optional provided part ID and an optional default viam.json, and returns a part ID to use.
+func resolvePartID(c *cli.Context, cloudJSON string) (string, error) {
+	partID := c.String(partFlag)
+	if len(partID) > 0 {
+		return partID, nil
+	}
+	if len(cloudJSON) == 0 {
+		return "", errors.New("no --part and no default json")
+	}
+	conf, err := rdkConfig.ReadLocalConfig(c.Context, cloudJSON, logging.Global())
+	if err != nil {
+		return "", err
+	}
+	return conf.Cloud.ID, nil
```

Comment:
`conf.Cloud` is a ptr. Should have a nil check

Reply:
done in d3b8041e